osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/PushSetupAlarmConverter.java

```diff
@@ -65,11 +96,38 @@ public PushSetupAlarm convertFrom(
     }
     final PushSetupAlarm.Builder builder = new PushSetupAlarm.Builder();
 
-    final String destination = source.getHost() + ":" + source.getPort();
-    final SendDestinationAndMethod sendDestinationAndMethod =
-        new SendDestinationAndMethod(
-            TransportServiceType.TCP, destination, MessageType.MANUFACTURER_SPECIFIC);
-    builder.withSendDestinationAndMethod(sendDestinationAndMethod);
+    if (source.getHost() != null && source.getPort() != null) {
+      final String destination = source.getHost() + ":" + source.getPort();
+      final SendDestinationAndMethod sendDestinationAndMethod =
+          new SendDestinationAndMethod(
+              TransportServiceType.TCP, destination, MessageType.MANUFACTURER_SPECIFIC);
```

Comment:
Is this still correct? 
According to the documentation (SMR5.2 P3 Companion v1.1) for Push Setup Alarm: 
> “message” is A-XDR encoded, value 0.

So shouldn't it then be `MessageType.A_XDR_ENCODED_X_DLMS_APDU`?

Reply:
Yes, that would make more sense.